code/game/machinery/autolathe.dm

```diff
@@ -268,8 +261,28 @@
 		busy = FALSE
 		return TRUE
 
+	if(panel_open)
+		balloon_alert(user, "close the panel first!")
+		return FALSE
+
 	return ..()
 
+/obj/machinery/autolathe/attackby_secondary(obj/item/O, mob/living/user, params)
+	if (busy)
+		to_chat(user, "<span class=\"alert\">The autolathe is busy. Please wait for completion of previous operation.</span>")
+		return TRUE
+
+	if(default_deconstruction_screwdriver(user, "autolathe_t", "autolathe", O))
+		return FALSE //returning this as FALSE prevents the screwdriver from being immediately eaten by the autolathe after you screw the panel open/closed. why? don't ask me
+
+	if(machine_stat)
+		return TRUE
+
+	if(panel_open)
+		balloon_alert(user, "close the panel first!")
+		return FALSE
+
+	return ..()
```

Comment:
You shouldn't return `..()` in secondary, since it'll call the normal attackby as well, which is a legacy behavior.

```suggestion
	return SECONDARY_ATTACK_CANCEL_CHAIN
```

Reply:
I think you meant SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN.
I tried it with the screwdriver step returning TRUE but it only stopped the screwdriver from being eaten on opening the panel. Closing the panel feeds the screwdriver in, so I kept the return FALSE.